packages/guardian-prover-health-check-ui/src/lib/dataFetcher.ts

```diff
@@ -0,0 +1,83 @@
+import { writable } from 'svelte/store';
+import type { Writable } from 'svelte/store';
+import { fetchGuardianProversFromContract } from './guardianProver/fetchGuardianProversFromContract';
+import { GuardianProverStatus, type Guardian, type GuardianProverIdsMap, type HealthCheck, type SignedBlocks, type SortedSignedBlocks } from './types';
+import { fetchSignedBlocksFromApi } from './blocks/fetchSignedBlocksFromApi';
+import { getGuardianProverIdsPerBlockNumber } from './blocks/getGuardianProverIdsPerBlockNumber';
+import { sortSignedBlocksDescending } from './blocks/sortSignedBlocks';
+import { publicClient } from './wagmi/publicClient';
+import { formatEther, type Address } from 'viem';
+import { fetchLatestGuardianProverRequest } from './guardianProver/fetchLatestGuardianProverRequest';
+
+export const guardianProvers: Writable<Guardian[]> = writable([]);
+export const apiResponse: Writable<HealthCheck[]> = writable([]);
+export const signedBlocks: Writable<SortedSignedBlocks> = writable([]);
+export const signerPerBlock: Writable<GuardianProverIdsMap> = writable({});
+
+export const lastGuardianFetchTimestamp: Writable<number> = writable(0);
+
+export async function fetchGuardians() {
+    const guardians = await fetchGuardianProversFromContract();
+    const guardiansWithBalance = [];
+    for (const guardian of guardians) {
+        const balance = await publicClient.getBalance({
+            address: guardian.address as Address
+        })
+
+        const balanceAsEther = formatEther(balance)
+        guardian.balance = balanceAsEther;
+
+        const status = await fetchLatestGuardianProverRequest(import.meta.env.VITE_GUARDIAN_PROVER_API_URL, guardian.id);
+        guardian.latestHealthCheck = status;
+
+        // if status.createdAt is older than 60 seconds, set guardian to dead
+        const createdAt = new Date(status.createdAt)
```

Comment:
@cyberhorsey have not checked if this something the stats will provide, but I assume if the last health check is of a certain age (>60s), we can consider the guardian dead.